examples/src/main/java/com/mapbox/navigation/examples/NavigationSettingsActivity.java

```diff
@@ -51,8 +59,29 @@ protected boolean isValidFragment(String fragmentName) {
     public void onCreate(final Bundle savedInstanceState) {
       super.onCreate(savedInstanceState);
       addPreferencesFromResource(R.xml.fragment_navigation_preferences);
+
       String gitHashTitle = String.format("Last Commit Hash: %s", BuildConfig.GIT_HASH);
       findPreference(getString(R.string.git_hash_key)).setTitle(gitHashTitle);
+
+      findPreference(getString(R.string.nav_native_history_retrieve_key)).setOnPreferenceClickListener(preference -> {
```

Comment:
Also are we doing the writing synchronously? It's going to block. We should fix in a future PR.

Reply:
Yes, it wasn't important for this quick iteration, but we can address in the future :+1: